### PR TITLE
Introduce document about core integrations

### PR DESCRIPTION
**Pull request by felixarntz**
I've started work on a document describing how we plan to integrate the Servehappy page with core. Please review, and then let's collaborate on it further through this PR.

### CODE REVIEW

File: CORE-INTEGRATIONS.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Core itself however should not use specific PHP versions as it should be possible to change the recommended version without updating WordPress. This may be reevaluated if there was an API exposing the recommended PHP version so that each WordPress install could gather that information remotely.
+
+## Maintenance Burden
+
+While updating the recommended version is only a small maintenance burden at this point, with the introduction of the Servehappy page and integration with core more areas will now reference that version specifically. A good idea would be to have a central location in the wordpress.org infrastructure to set that version. It could be a simple setting, and there could be a shortcode to include that version anywhere on wordpress.org. In addition, the wordpress.org API could also include that version somewhere which would allow WordPress core to be more specific about upgrade recommendations as well.
```
Comment (joyously) [inline CORE-INTEGRATIONS.md]: Having the recommended PHP version centrally makes sense, but wouldn't it be WP version specific, meaning that you'd need more than one? It's not that it changes much, but it _would_ change.

Comment (grappler) [reply CORE-INTEGRATIONS.md]: 👍 As WP 4.8 does not work with PHP 7.2. So we can't recommend people to use PHP 7.2 with it in the future when we are on WP 5.8.
WP 4.4 added support for PHP 7.0 and WP 4.7 added support for PHP 7.1

Comment (JDGrimes) [reply CORE-INTEGRATIONS.md]: The recommended version on the requirements page only refers to the most recent version of WP. 

However, it is true that when/if we are showing notices to users on older versions of WP, the recommended version will be different. Really though, wouldn't it be best to have them upgrade WP *first*, and then upgrade PHP?

**Comment by schlessera** [comment]
Merging this as a first draft. Changes and discussions can then happen as PRs.

**Pull request merged**